### ISSUE TITLE
Augumentation doesn't sync box and image

### ISSUE THREAD

**Issue by wassname** [fizyr/keras-retinanet#150]
In my testing, augmentation doesn't flip box and image together. Here is a notebook showing/plotting it https://gist.github.com/wassname/970c914745d49bd5999d9896e2933db2.

The problem is in keras, and my [keras PR](https://github.com/fchollet/keras/pull/7613) from months ago fixes it (I've tested this). But it's waiting on a review so if you want this fixed head over and give it a review. All I did is add random.seed(seed) where needed.

**Comment by hgaiser** [comment]
Thanks for looking into this. I'll have to look at this later so that I can give it some more attention. All I can say at the moment is that the RNG part in Keras is bit of a mess, but I think you've noticed that too ;)

I'll hopefully have some time to check this somewhere this week.

**Comment by wassname** [comment]
Yeah the RNG part is doing my head in. No hurry, there's more important things than data augmentation.

**Comment by de-vri-es** [comment]
This could be fixed together with #68. I had some initial work on it in the `cheap-augmentation` branch but I have other priorities right now so I don't know when I'll continue on it. If someone wants to continue from that branch, feel free.

**Comment by wassname** [comment]
Nice I'll check out that branch, not sure if I'll continue it (depends on how it fits into my current project), but good to know

**Comment by wassname** [comment]
Not sure if you've seen torchvision, but their [transforms.py](https://github.com/pytorch/vision/blob/master/torchvision/transforms/transforms.py) is tidy and nice and uses PIL. It's similar to your approach of making modular transforms although it's not able to masks and images together.

**Comment by hgaiser** [comment]
Actually you had a small bug in the plotting code, the fixed version is [here](https://gist.github.com/hgaiser/1fcafaf7809cbd1e0f3797f1914e5db3). The order of the boxes is `(x1, y1, x2, y2)`, but you used it as `(y1, x1, y2, x2)`.

**Comment by wassname** [comment]
Thanks for taking a look. 

That might be an error I made, but it can't be the only cause. If that was the cause then my annotations would be consistently flipped in the x and y axis, but they aren't. 

Still this shows augmentation it works for you, which means it's hard to reproduce. I'll try it again with the latest code/keras to check it's still a problem. If you have time could you send me a `pip freeze` so I can compare our python environments please?

**Comment by awilliamson** [comment]
wassname, it might also be worth choosing different bbox coords than 0, 0, 2, 2 just so any artefacts in the flipping can pop out. Using boundary numbers isn't always a good idea for minimal working examples.

**Comment by wassname** [comment]
I tried again with keras 2.1.2 and the latest retinanet and it all worked [fine](https://gist.github.com/wassname/970c914745d49bd5999d9896e2933db2). Sorry about this everyone. 

I think it might have been a keras 2.0.9 vs 2.1.2 issue since this appears to have fixed some augmentation repeatability issues.
